data_linter/validation.py

```diff
@@ -273,28 +273,24 @@ def _read_data_and_validate(
         metadata (dict): The metadata for the table
     """
     print(f"Reading and validating: {filepath}")
-    dialect = dialects.Dialect()
-    skip_checks = []
+    
+    skip_errors = []
+    
+    # assert the correct dialect and checks
+    if metadata["data_format"] == "json":
+        expected_headers = [c["name"] for c in metadata["columns"] if c not in metadata.get("partitions", [])]
+        dialect = dialects.JsonDialect(keys=expected_headers)
+    else: # assumes CSV
+        dialect = dialects.Dialect(header_case=table_params.get("headers-ignore-case", False))
+        if table_params.get("expect-header"):
```

Comment:
shouldn't this be `if not table_params.get("expect-header"):`

Reply:
It should!